### PR TITLE
ES6: Migrated "auth" directory

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

File: app/script/auth/AuthRepository.js [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+      }
+    }
+    );
+  }
+
+  // Get the cached access token from the Amplify store.
+  get_cached_access_token() {
+    return new Promise(((resolve, reject) => {
+      let access_token = z.util.StorageUtil.get_value(z.storage.StorageKey.AUTH.ACCESS_TOKEN.VALUE);
+      let access_token_type = z.util.StorageUtil.get_value(z.storage.StorageKey.AUTH.ACCESS_TOKEN.TYPE);
```
Comment (herrmannplatz) [inline app/script/auth/AuthRepository.js]: const

File: app/script/auth/AuthRepository.js [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+   *  access_token_ttl: 900000 => 900s/15min
+   *
+   * @param {Object, String} access_token_data - Access Token
+   * @option {String} access_token_data - access_token
+   * @option {String} access_token_data - expires_in
+   * @option {String} access_token_data - type
+   */
+  save_access_token(access_token_data) {
+    let expires_in_millis = 1000 * access_token_data.expires_in;
+    let expiration_timestamp = Date.now() + expires_in_millis;
```
Comment (herrmannplatz) [inline app/script/auth/AuthRepository.js]: const

File: app/script/auth/AuthRepository.js [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+   * @private
+   * @param {Object, String} access_token_data - Access Token
+   * @option {String} access_token_data - access_token
+   * @option {String} access_token_data - expires_in
+   * @option {String} access_token_data - type
+   * @param {Integer} expiration_timestamp - Timestamp when access token expires
+   */
+  _log_access_token_update(access_token_data, expiration_timestamp) {
+    let expiration_log = z.util.format_timestamp(expiration_timestamp, false);
+    return this.logger.info(`Saved updated access token. It will expire on: ${expiration_log}`, access_token_data);
```
Comment (herrmannplatz) [inline app/script/auth/AuthRepository.js]: const

File: app/script/auth/AuthService.js [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+'use strict';
+
+(function() {
+  window.z = window.z || {};
+  window.z.auth = z.auth || {};
+
+  const POST_ACCESS = {
+    RETRY_LIMIT: 10,
+    RETRY_TIMEOUT: 500,
+  };
```
Comment (herrmannplatz) [inline app/script/auth/AuthService.js]: could be a static property as well or?

File: app/script/auth/AuthService.js [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+     * @note Don't use our client wrapper here, because to query "/access" we need to set "withCredentials" to "true" in order to send the cookie.
+     * @see https://staging-nginz-https.zinfra.io/swagger-ui/#!/auth/authenticate
+     * @param {Integer} retry_attempt - Retry attempts when a request fails
+     * @returns {Promise}
+     */
+    post_access(retry_attempt = 1) {
+      return new Promise((resolve, reject) => {
+        this.client.request_queue_blocked_state(z.service.RequestQueueBlockedState.ACCESS_TOKEN_REFRESH);
+
+        let config = {
```
Comment (herrmannplatz) [inline app/script/auth/AuthService.js]: could be const as well

File: app/script/auth/AuthRepository.js [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+  register(new_user) {
+    return this.auth_service.post_register(new_user).then((response) => {
+      z.util.StorageUtil.set_value(z.storage.StorageKey.AUTH.PERSIST, true);
+      z.util.StorageUtil.set_value(z.storage.StorageKey.AUTH.SHOW_LOGIN, true);
+      z.util.StorageUtil.set_value(new_user.label_key, new_user.label);
+      this.logger.info(`COOKIE::'${new_user.label}' Saved cookie label with key '${new_user.label_key}' in Local Storage`, {
+        key: new_user.label_key,
+        value: new_user.label,
+      }
+        );
```
Comment (lipis) [inline app/script/auth/AuthRepository.js]: something is off here..

File: app/script/auth/AuthRepository.js [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+      this.logger.info(`Scheduling next access token refresh for '${time}'`);
+
+      return this.access_token_refresh = window.setTimeout(() => {
+        if (callback_timestamp > (Date.now() + 15000)) {
+          return this.logger.info(`Access token refresh scheduled for '${time}' skipped because it was executed late`);
+        } else if (navigator.onLine) {
+          return this.renew_access_token(`Schedule for '${time}'`);
+        } else {
+          return this.logger.info(`Access token refresh scheduled for '${time}' skipped because we are offline`);
+        }
```
Comment (lipis) [inline app/script/auth/AuthRepository.js]: no else returns

Comment (herrmannplatz) [reply app/script/auth/AuthRepository.js]: http://eslint.org/docs/rules/no-else-return

Comment (lipis) [reply app/script/auth/AuthRepository.js]: let's do this :)

File: app/script/auth/AuthService.js [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+            this.logger.error(`Requesting access token failed after ${retry_attempt} attempt(s): ${errorThrown}`, jqXHR);
+            reject(new z.auth.AccessTokenError(z.auth.AccessTokenError.TYPE.REQUEST_FORBIDDEN));
+          }
+
+          if (retry_attempt <= AuthService.POST_ACCESS.RETRY_LIMIT) {
+            retry_attempt++;
+
+            const _retry = () => {
+              return this.post_access(retry_attempt).then(resolve).catch(reject);
+            };
```
Comment (ffflorian) [inline app/script/auth/AuthService.js]:
```js
const _retry = () => this.post_access(retry_attempt).then(resolve).catch(reject);
```

File: app/script/auth/AuthService.js [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+          headers: {
+            'Content-Encoding': 'gzip',
+          },
+          processData: false,
+          type: 'POST',
+          url: `${this.client.create_url(`${AuthService.URL_REGISTER}?challenge_cookie=true`)}`,
+          xhrFields: {
+            withCredentials: true,
+          },
+        }).done((data) => {
```
Comment (ffflorian) [inline app/script/auth/AuthService.js]:
```js
})
.done(...
```

File: app/script/auth/AuthService.js [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+          resolve(data);
+        }).fail((jqXHR, textStatus, errorThrown) => {
```
Comment (ffflorian) [inline app/script/auth/AuthService.js]:
```js
})
.fail(...
```

**Pull request merged**